config/default.yml

```diff
@@ -30,6 +30,10 @@ Metrics:
 
 Metrics/BlockLength:
   Max: 25
+  Exclude:
+    - spec/
```

Comment:
I agree there is a problem, but before we merge this, I'd like to understand
why the config for `Metrics` does not apply to `Metrics/BlockLength`, isn't it
supposed to behave this way? Should we remove it on `Metrics` first?

https://github.com/alpinelab/codestyle/blob/588a78d14775214cc3619463c0499047a03269b0/config/default.yml#L27-L29

---

Does rubocop handles `spec/` differently than `spec`? If the answer is no, I
think we should favor just `spec`.

Reply:
I think @Polo2 had the issue in files ending with `spec/**/*.rb` files, because
I noticed and reported the same issue some time ago (not on GitHub however).

Reply:
I did investigated :male_detective: !
Issue don't comes from `spec/**/*.rb` to `spec/`, issue comes from how keys are merged:

On `rails.yml`, I realized that:
```ruby
Metrics/BlockLength:
  Exclude:
    - config/routes.rb
```
overrides informations wich comes from on `config.yml/`:
```ruby
Metrics:
  Exclude:
    - spec/**/*.rb
```

I don't know how to fix this .... Could it come from: 
```ruby
inherit_mode:
  merge:
    - Exclude
```
?

Reply:
Well,
```
inherit_mode:
  merge:
    - Exclude
```
makes Rubocop merges values passed to `Exclude`.

Here, it might not be handled by inheritance but by because `Metrics` is the namespace that contains `Metrics/BlockLength`.

I can't find anything in the documentation about that, but what about trying a cop-specific inheritance mode like:

```yaml
Metrics/BlockLength:
  inherit_mode:
    merge:
      - Exclude
  Exclude:
    - config/routes.rb
```

If it does not work, you may have to use the not-very-DRY version:

```yaml
Metrics/BlockLength:
  Exclude:
    - spec/**/*.rb
    - config/routes.rb
```